pkg/app/piped/platformprovider/terraform/terraform.go

```diff
@@ -158,12 +161,99 @@ type PlanResult struct {
 	Adds     int
 	Changes  int
 	Destroys int
+
+	PlanOutput string
 }
 
 func (r PlanResult) NoChanges() bool {
 	return r.Adds == 0 && r.Changes == 0 && r.Destroys == 0
 }
 
+func (r PlanResult) Render() string {
+	terraformDiffStart := "Terraform will perform the following actions:"
+	terraformDiffEnd := "─"
+
+	startIndex := strings.Index(r.PlanOutput, terraformDiffStart) + len(terraformDiffStart)
+	endIndex := strings.Index(r.PlanOutput, terraformDiffEnd)
+	out := r.PlanOutput[startIndex:endIndex]
+
+	rendered := ""
+	curlyBracketStack := stack.New()
+	squareBracketStack := stack.New()
+
+	scanner := bufio.NewScanner(strings.NewReader(out))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+
+		r := []rune(line)
```

Comment:
This implementation is a little bit complicated compared to what we want to do.
I'm glad if I could hear why we need to check a character one by one.